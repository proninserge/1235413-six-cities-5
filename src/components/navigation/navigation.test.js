import renderer from 'react-test-renderer';
import Navigation from './navigation';

const noop = () => {};

describe(`<Navigation /> render`, () => {

  it(`correctly renders Navigation if authorized`, () => {
    const tree = renderer
      .create(
          <Navigation
            email={`johndoe@yahoo.com`}
            isAuthorized={true}
            onNavigationClick={noop}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`correctly renders Navigation if not authorized`, () => {
    const tree = renderer
      .create(
          <Navigation
            email={`johndoe@yahoo.com`}
            isAuthorized={false}
            onNavigationClick={noop}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

