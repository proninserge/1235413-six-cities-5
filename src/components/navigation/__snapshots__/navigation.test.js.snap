// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> render correctly renders Navigation if authorized 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="#"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__user-name user__name"
        >
          johndoe@yahoo.com
        </span>
      </a>
    </li>
  </ul>
</nav>
`;

exports[`<Navigation /> render correctly renders Navigation if not authorized 1`] = `
<nav
  className="header__nav"
>
  <ul
    className="header__nav-list"
  >
    <li
      className="header__nav-item user"
    >
      <a
        className="header__nav-link header__nav-link--profile"
        href="#"
        onClick={[Function]}
      >
        <div
          className="header__avatar-wrapper user__avatar-wrapper"
        />
        <span
          className="header__login"
        >
          Sign in
        </span>
      </a>
    </li>
  </ul>
</nav>
`;
